### PR TITLE
debuginfo: Prevent redundant debug info uploads

### LINKED ISSUE #716: debuginfo: Prevent redundant uploads with a proper error
I think primarily I'd err on the side of not uploading if we don't absolutely have to because we would be uploading the same debuginfos thousands of times most likely.

__Originally posted by @brancz in https://github.com/parca-dev/parca/issues/705#issuecomment-1060318529__

### PR DESCRIPTION
**Pull request by kakkoyun**
Signed-off-by: Kemal Akkoyun <kakkoyun@gmail.com>

Fixes #716

### CODE REVIEW

File: pkg/debuginfo/store.go [hunk 1]
```diff
@@ -198,6 +200,20 @@ func validateID(id string) error {
 	return nil
 }
 
+func (s *Store) find(ctx context.Context, key string) (bool, error) {
+	found := false
+	err := s.bucket.Iter(ctx, key, func(_ string) error {
```
Comment (brancz) [inline pkg/debuginfo/store.go:205]: I'm happy for this to be a follow-up, but I expected that we also check the upstream debug info endpoint here.

**Review by brancz — APPROVED** [review]
Nice! (not sure why but I had assumed we already did this so great find!)

File: pkg/debuginfo/store.go (hunk 1 above)
Comment (kakkoyun) [reply pkg/debuginfo/store.go:205]: To be crystal clear here, by the “the upstream debug info endpoint” you meant to look up the debuginfod servers that have been specified? 

Or do you want to utilize existing expose `Exists` API? 

For the first one, we check it at access time:

https://github.com/parca-dev/parca/blob/36b893ae5eb7adfe0ee28c34020b4d64fe123ced/pkg/debuginfo/store.go#L252-L260

Comment (brancz) [reply pkg/debuginfo/store.go:205]: I did indeed mean the first one, thanks for clarifying! Ignore my comment then!

**Pull request merged**